app/javascript/packs/leafletMap.js

```diff
@@ -114,38 +127,54 @@ async function getView() {
     });
 }
 
-function addIndoorLabel(feature, layer) {
-    layer.bindTooltip(feature.properties.name, {
-        permanent: true,
-        direction: "center",
-        className: "indoor-label",
-    });
+function setupGeoJsonFeature(feature, layer) {
+    const level = parseInt(feature.properties.level_name);
+    if (Number.isInteger(level)) {
+        if (!floors[level]) {
+            floors[level] = { rooms: L.layerGroup(), labels: L.layerGroup() };
+            const layer = L.layerGroup([floors[level].rooms, floors[level].labels]);
```

Comment:
Choose another name for this variable as `layer` is already defined in the outer scope.